src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -0,0 +1,489 @@
+    return lastFocused ? lastFocused->_profile : std::nullopt;
+}
+
+// Method Description:
+// - Returns true if this pane was the last pane to be focused in a tree of panes.
+// Arguments:
+// - <none>
+// Return Value:
+// - true iff we were the last pane focused in this tree of panes.
+bool Pane::WasLastFocused() const noexcept
+{
+    return _lastFocused;
+}
+
+// Method Description:
+// - Returns true iff this pane has no child panes.
+// Arguments:
+// - <none>
+// Return Value:
+// - true iff this pane has no child panes.
+bool Pane::_IsLeaf() const noexcept
+{
+    return _splitState == SplitState::None;
+}
+
+// Method Description:
+// - Returns true if this pane is currently focused, or there is a pane which is
+//   a child of this pane that is actively focused
+// Arguments:
+// - <none>
+// Return Value:
+// - true if the currently focused pane is either this pane, or one of this
+//   pane's descendants
+bool Pane::_HasFocusedChild() const noexcept
+{
+    const bool controlFocused = _control != nullptr &&
+                                _control.GetControl().FocusState() != FocusState::Unfocused;
+    const bool firstFocused = _firstChild != nullptr && _firstChild->_HasFocusedChild();
+    const bool secondFocused = _secondChild != nullptr && _secondChild->_HasFocusedChild();
+
+    return controlFocused || firstFocused || secondFocused;
+}
+
+// Method Description:
+// - Update the focus state of this pane, and all it's descendants.
+//   * If this is a leaf node, and our control is actively focused, well mark
+//     ourselves as the _lastFocused.
+//   * If we're not a leaf, we'll recurse on our children to check them.
+void Pane::UpdateFocus()
+{
+    if (_IsLeaf())
+    {
+        const bool controlFocused = _control != nullptr &&
+                                    _control.GetControl().FocusState() != FocusState::Unfocused;
+
+        _lastFocused = controlFocused;
+    }
+    else
+    {
+        _lastFocused = false;
+        _firstChild->UpdateFocus();
+        _secondChild->UpdateFocus();
+    }
+}
+
+// Method Description:
+// - Focuses this control if we're a leaf, or attempts to focus the first leaf
+//   of our first child, recursively.
+void Pane::_FocusFirstChild()
+{
+    if (_IsLeaf())
+    {
+        _control.GetControl().Focus(FocusState::Programmatic);
+    }
+    else
+    {
+        _firstChild->_FocusFirstChild();
+    }
+}
+
+// Method Description:
+// - Attempts to update the settings of this pane or any children of this pane.
+//   * If this pane is a leaf, and our profile guid matches the parameter, then
+//     we'll apply the new settings to our control.
+//   * If we're not a leaf, we'll recurse on our children.
+// Arguments:
+// - settings: The new TerminalSettings to apply to any matching controls
+// - profile: The GUID of the profile these settings should apply to.
+void Pane::CheckUpdateSettings(const TerminalSettings& settings, const GUID& profile)

```

Comment:
@zadjii-msft nit: This name is a little confusing.  Check doesn't convey "update the settings".

Reply:
That's a good point.